File: 2033-minimum-operations-to-make-a-uni-value-grid/2033-minimum-operations-to-make-a-uni-value-grid.py
class Solution:
    def minOperations(self, grid: List[List[int]], x: int) -> int:
        nums = []
        for row in grid:
            nums.extend(row)

        remainder = nums[0] % x
        if any(num % x != remainder for num in nums):
            return -1

        nums.sort()
        median = nums[len(nums) // 2]
        operations = 0
        for num in nums:
            operations += abs(num - median) // x
        
        return operations

File: 2033-minimum-operations-to-make-a-uni-value-grid/2033-minimum-operations-to-make-a-uni-value-grid.cpp
class Solution {
public:
    int minOperations(vector<vector<int>>& grid, int x) {
        vector<int> vals;
        vals.reserve(grid.size() * grid[0].size());

        int r = grid[0][0] % x;

        for (auto& row : grid) {
            for (int v : row) {
                if (v % x != r) return -1;
                vals.push_back(v);
            }
        }

        nth_element(vals.begin(), vals.begin() + vals.size() / 2, vals.end());
        int median = vals[vals.size() / 2];

        int ops = 0;
        for (int v : vals) {
            ops += abs(v - median) / x;
        }

        return ops;
    }
};